### ISSUE TITLE
ref by class name ?

### ISSUE THREAD

**Issue by momala454** [zircote/swagger-php#1311]
Is it possible to reference a schema using the class name instead of the schema name ?

the following doesn't work obviously as it doesn't return something in the form of "#/components/schemas/foo" when doing Foo::class

```php
(...)
new OA\Schema(
                        properties: [new OA\Property(
                            property: 'file_profile',
                            ref: Foo::class
                        )]
                    )


namespace helloWorld;
#[\OpenApi\Attributes\Schema()
class foo
{
}
```

**Comment by DerManoMann** [comment]
It's something that is not consistent yet, but using type instead of ref might work

**Comment by momala454** [comment]
thanks

**Comment by momala454** [comment]
With my use case, it seems to work with requestBody, but not with responses. I saw you're refactoring it in https://github.com/zircote/swagger-php/issues/1310, great !

**Comment by likemusic** [comment]
@momala454  Make sure that `\MyNamespace\Class` has `OA\Schema` attribute.

**Comment by momala454** [comment]
> 

it doesn't have #[\OA\Schema] but it have an extended Attribute :

```php
#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::TARGET_METHOD | \Attribute::TARGET_PROPERTY)]
class ApiExceptionAttribute extends \OpenApi\Attributes\Schema
{}
```

It doesn't work when I do it like this. Using "type" works only if I use directly `\OpenApi\Attributes\Schema`

**Comment by likemusic** [comment]
> It doesn't work when I do it like this. Using "type" works only if I use directly \OpenApi\Attributes\Schema

@momala454 I have the same issue yesterday)

Look at here: https://github.com/zircote/swagger-php/blob/4f63dcbdd216501125870b409139ff7324f3703e/src/Processors/ExpandClasses.php#L26

Processor that expand classes for schemas do this just for `OA\Schema` and `OAT\Schema` (`true` flag at `$analysis->getAnnotationsOfType()` call) because many other annotation classes inherited from Schema class and processed separately by other processors.

![image](https://user-images.githubusercontent.com/603401/191298729-c810a392-eb04-49d4-bd3a-246074fb17b2.png)

At the moment I think that the best way to fix it would be add $exclude (array) parameter for `getAnnotationsOfType()` method to be able to extract schemas with child classes except already defined in package.

**Comment by momala454** [comment]
Changing the `true` strict parameter to false in the class `ExpandClasses` is not enough to fix it

**Comment by likemusic** [comment]
@momala454 You are rigth. Problem was in another processor. See https://github.com/zircote/swagger-php/pull/1315

You can use it untill it woud be merged with fixes in `composer.json` like below:
```
    "require": {
        "zircote/swagger-php": "dev-fix-schemas-inheritance-v2 as 4.4.9"
    },
    "repositories": [
        {
            "type": "vcs",
            "url": "git@github.com:likemusic/swagger-php.git"
        }
    ]
...
```

**Comment by momala454** [comment]
@likemusic , your second fix doesn't work neither for me, it is changing all my "refs" into "type", where the value of type becomes `MyNamespace\\Class`

Actually, not all, but most of them. I don't know what makes some different than others, but it doesn't fix the value when i use "type"

**Comment by DerManoMann** [comment]
The original issue was fixed in #1418 Any other issues should be raised in new, separate tickets.
